dotnet/samples/KernelSyntaxExamples/Example57_FunctionEventHandlers.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Events;
+using RepoUtils;
+
+#pragma warning disable RCS1214 // Unnecessary interpolated string.
+
+// ReSharper disable once InconsistentNaming
+public static class Example57_FunctionEventHandlers
+{
+    private static string? openAIModelId;
+    private static string? openAIApiKey;
+
+    public static async Task RunAsync()
+    {
+        Console.WriteLine("\n======== Using Function Execution Handlers ========\n");
+
+        openAIModelId = TestConfiguration.OpenAI.ModelId;
```

Comment:
nit: We should make it easier for our examples to leverage azure by default if available and then fallback to openai.